src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1017,35 +1020,75 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
             case TSet(_) => TSet(t)
           }
           case error =>
-            parseError(s"method `$method' expects a lambda function [param => Any], got invalid mapping [param => $error]")
+            parseError(s"method `$method' expects a lambda function (param => Any), got invalid mapping (param => $error)")
+        }
+
+      case (agg: TAggregable, "map", rhs) =>
+        val (param, body) = rhs match {
+          case Array(Lambda(_, p, b)) => (p, b)
+        }
```

Comment:
I want to check that it's a lambda function, though.  I think the code you suggested will produce bad error messages